numpy/core/src/multiarray/descriptor.c

```diff
@@ -135,12 +124,10 @@ _try_convert_from_dtype_attr(PyObject *obj)
         return NULL;
     }
 
-    PyArray_Descr *newdescr;
-    int ret = PyArray_DescrConverter(dtypedescr, &newdescr);
-
+    PyArray_Descr *newdescr = _convert_from_any(dtypedescr, 0);
```

Comment:
I could make this respect alignment by adding an `align` argument to `_try_convert_from_dtype_attr`, but we'd then need to plumb that through `PyArray_DescrFromTypeObject` too, and the whole thing wouldn't make much sense anyway - user types are unlikely to have separate aligned and unaligned variants.

At any rate, this matches the old behavior.

Reply:
OK, lets keep old behaviours for now in any case. If you wanted, we could add a comment, or even issue. But my guess is that it is not worth the trouble right now.